### PR TITLE
add log history tracking and ack function

### PR DESCRIPTION
**Pull request by tsloughter**

### CODE REVIEW

File: src/vg_active_segment.erl [hunk 1]
```diff
@@ -29,18 +30,33 @@
...
+            ok
+    end,
+
     gen_server:start_link(?SERVER(Topic, Partition), ?MODULE, [Topic, Partition, NextBrick], []).
 
 write(Topic, Partition, MessageSet) ->
     gen_server:call(?SERVER(Topic, Partition), {write, MessageSet}).
 
+ack(Topic, Partition, LatestId) ->
+    ets:select_delete(?TAB(Topic, Partition), [{{'$1', '$2'},
```
Comment (evanmcc) [inline src/vg_active_segment.erl]: We're going to be calling this a lot, is there a way to do this with a straight delete?

**Review by evanmcc — APPROVED** [review]
leaves an open question: what's the paper say wrt what to do with these writes on crash and recovery?

File: src/vg_active_segment.erl (hunk 1 above)
Comment (tsloughter) [reply src/vg_active_segment.erl]: Why? What is a "straight delete"?

Comment (evanmcc) [reply src/vg_active_segment.erl]: http://erlang.org/doc/man/ets.html#delete-2

I feel like I've spent a lot of time wrestling with select_* related performance problems.  Not a big deal right now.

Comment (tsloughter) [reply src/vg_active_segment.erl]: But we need all the keys. I don't know of a way to get all the records to delete without a select, so even if we did a delete with each key we'd need to first select them.

**Comment by tsloughter** [comment]
The papers won't have anything on our type of node recovery, since they recover as the tail and we stay in the same position.

**Pull request merged**